scripts/ci/processCoverageReport.py

```diff
@@ -0,0 +1,414 @@
+
+            if result.returncode != 0:
+                print(f"Error getting changed files with second approach: {result.stderr}")
+                return []
+
+        changed_files = [file for file in result.stdout.splitlines() if file.endswith('.java')]
+
+        print(f"Found {len(changed_files)} changed Java files")
+        if changed_files:
+            print("Changed Java files:")
+            for file in changed_files[:10]:
+                print(f"  - {file}")
+
+            if len(changed_files) > 10:
+                print(f"  ... and {len(changed_files) - 10} more files")
+
+        return changed_files
+    except Exception as e:
+        print(f"ERROR: Failed to get changed files: {str(e)}")
+        return []
+
+def match_changed_files_to_coverage(changed_files, coverage_data):
+    """Match changed files to coverage data."""
+    print("Matching changed files to coverage data...")
+    matches = []
+    unmatched = []
+
+    changed_file_names = [(f, os.path.basename(f).lower()) for f in changed_files]
+    print(f"Processing {len(changed_files)} changed files against {len(coverage_data)} coverage reports")
+
+    for changed_file, short_name in changed_file_names:
+        found = False
+
+        for report_index, report_coverage in enumerate(coverage_data):
+            files_coverage = report_coverage.get('files', {})
+
+            for coverage_path, coverage_info in files_coverage.items():
+                if changed_file.replace('\\', '/') in coverage_path:
+                    matches.append((changed_file, coverage_info))
+                    found = True
+                    print(f"Match found for {changed_file} in report #{report_index+1}")
+                    break
+
+                if short_name == os.path.basename(coverage_path).lower():
+                    matches.append((changed_file, coverage_info))
+                    found = True
+                    print(f"Match found for {short_name} in report #{report_index+1}")
+                    break
+
+            if found:
+                break
+
+        if not found:
+            unmatched.append(changed_file)
+
+    print(f"Matched {len(matches)} of {len(changed_files)} changed files to coverage data")
+
+    if unmatched:
+        print("Unmatched files:")
+        for file in unmatched:
+            print(f"  - {file}")
+
+    return matches
+
+def calculate_overall_coverage(coverage_data, counter_type='LINE'):
+    """Calculate overall coverage from all reports."""
+    print(f"Calculating overall {counter_type} coverage from {len(coverage_data)} reports...")
+    total_covered = 0
+    total_missed = 0
+
+    for data in coverage_data:
+        if 'overall' in data and counter_type in data['overall']:
+            total_covered += data['overall'][counter_type]['covered']
+            total_missed += data['overall'][counter_type]['missed']
+
+    if total_covered + total_missed > 0:
```

Comment:
Since `total_missed >= 0` then you can do:
```
overall_coverage = 0 if total_covered == 0 else (total_covered / (total_covered + total_missed)) * 100
```
I'd suggest just using total_lines everywhere though, because then this becomes:
```
overall_coverage = 0 if total_lines == 0 else 100 * total_covered / total_lines
```